crypto/tbs/src/lib.rs

```diff
@@ -282,10 +282,9 @@ mod tests {
         let sigs = sks
             .iter()
             .enumerate()
-            .map(|(idx, sk)| (idx, sign_blinded_msg(bmsg, *sk)))
-            .collect::<Vec<_>>();
```

Comment:
I think this change will make it `n` times slower. Tbf, it's probably bad API design from my side.

`combine_valid_shares` takes an `IntoIter` argument where `IntoIter::Iter: Clone` and uses it as a random access iterator by cloning the iter and calling `nth`. Afaik stable rust doesn't have a `RandomAccessIterator` trait, maybe I should try to impl it (either locally or upstream).

Long story short, you are probably doing a 4ms sign operation over and over again by not collecting here.

Reply:
Or I could just require a slice, but that felt lame xD

Reply:
I think I understand...please see the latest commit to see if it solves the issue.